core/src/main/java/fr/sncf/osrd/api/pathfinding/response/PathWaypointResult.java

```diff
@@ -19,11 +19,11 @@ public class PathWaypointResult {
      */
     public boolean suggestion;
     /**
-     * Track the point is on
+     * Track location
      */
-    public String track;
+    public PathWaypointLocation location;
     /**
-     * Offset of the point on the track
+     * Waypoint position on the path
      */
     public double position;
```

Comment:
I really don't like that the field type and name stay the same, but what it is changes.
Can you please rename it to something like `path_offset`?